### ISSUE TITLE
Input tensor to yolov8

### ISSUE THREAD

**Issue by MindSystemm** [ultralytics/ultralytics#8631]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello, 

I'm working on adversarial training and to do so, I need my input to the yolo model to be differentiable. To do so, I'm inputting a tensor to the model. However, I have difficulties to interpret the output. In the documentation, it's written that it's possible to get the confidence from the model using results.boxes.conf but it doesn't seem to work with tensors. Here's my code 

```python
image_url = "XXXX"
image = Image.open(image_url)
transform = transforms.Compose([
    transforms.Resize((640, 640)),
    transforms.ToTensor(),
])
image_tensor = transform(image)
image_tensor = image_tensor.unsqueeze(0)
image_tensor.requires_grad_(True)

model_original = YOLO('custom.pt')
model_original.to(device='cpu')
model_original.training = False

results = model(image_tensor)
for r in results:
    print(r.boxes)
```

The error I'm getting is the following : 

![Error](https://github.com/ultralytics/ultralytics/assets/26023804/710d275c-e09c-409f-b761-300e94c5c1d2)

Thanks for support

**Comment by MindSystemm** [comment]
This can be solved by resizing the picture to a size divisible by 32 
```python 
image = Image.open(image_url)
transform = transforms.Compose([
    transforms.Resize((640, 640)),
    transforms.ToTensor(),
])
image_tensor = transform(image)
image_tensor = image_tensor.unsqueeze(0)
image_tensor.requires_grad_(True)
```

**Comment by glenn-jocher** [comment]
@MindSystemm hello! 👋

It looks like you're on the right track with your approach. When you input a tensor directly to the model and wish to access the results, ensure you're accessing the attributes correctly. The output `results` from `model(image_tensor)` is a list of `Results` objects, so you should access the attributes of these objects directly. Here's a slight modification to your loop for printing the confidence scores:

```python
results = model_original(image_tensor)
for r in results:
    print(r.boxes.conf)
```

However, based on the error screenshot you provided (which unfortunately didn't load here), if you're encountering an issue specifically related to accessing `.conf` or any other attribute, it might be due to how the results are being processed or a mismatch in expected output format.

If the issue persists, could you please share more details about the error message or clarify the exact problem you're facing with accessing the confidence scores? This will help in providing a more precise solution. 🛠️

Keep up the great work with adversarial training!

**Comment by MindSystemm** [comment]
> @MindSystemm hello! 👋
> 
> It looks like you're on the right track with your approach. When you input a tensor directly to the model and wish to access the results, ensure you're accessing the attributes correctly. The output `results` from `model(image_tensor)` is a list of `Results` objects, so you should access the attributes of these objects directly. Here's a slight modification to your loop for printing the confidence scores:
> 
> ```python
> results = model_original(image_tensor)
> for r in results:
>     print(r.boxes.conf)
> ```
> 
> However, based on the error screenshot you provided (which unfortunately didn't load here), if you're encountering an issue specifically related to accessing `.conf` or any other attribute, it might be due to how the results are being processed or a mismatch in expected output format.
> 
> If the issue persists, could you please share more details about the error message or clarify the exact problem you're facing with accessing the confidence scores? This will help in providing a more precise solution. 🛠️
> 
> Keep up the great work with adversarial training!

Indeed it works, thanks ! The next step for adversarial training would be to have access to the logits of the model. Is it natively possible with YOLO or does it require modification of the model return arguments?

**Comment by glenn-jocher** [comment]
Hey @MindSystemm, glad to hear the solution worked for you! 😊 For accessing the logits directly from YOLOv8, the model does not natively return them in its default configuration. However, you can modify the model's forward method to include logits in the return statement. This requires a bit of tweaking in the source code. If you're comfortable with diving into the code, look for the final layer before the output and adjust accordingly to return the logits. If you need further assistance with this, feel free to ask! Keep pushing the boundaries with your adversarial training experiments! 🚀
